generate_mesh.py

```diff
@@ -0,0 +1,32 @@
+import numpy as np
+
+
+def generate_mesh(layer_outer_radii, nodes_per_layer):
+    """ This function generates an array of grid points in the radial direction,
+    or mesh.
+    Grid points are spaced evenly between the previous outer radii
+    (zero for the first layer) and the specified outer radii.
+    generate_mesh returns a single array of r values
+
+    inputs
+    ------
+    - layer_outer_radii : list of radii for each layer of the fuel particle.
+    The first layer is assumed to start at r=0
+    - nodes_per_layer : list with the number of grid points
+      corresponding to each layer
+
+    outputs
+    -------
+    - mesh : a list of grid points
+    """
+
+    mesh = np.zeros(sum(nodes_per_layer))
+    nodes_sum = 0

```

Comment:
perhaps `layer_first_node` and `layer_last_node` communicate better the purpose of `nodes_sum` and `nodes_sum_new`